app/src/main/java/com/github/swent/echo/compose/authentication/createProfile.kt

```diff
@@ -0,0 +1,251 @@
+import androidx.navigation.compose.rememberNavController
+import com.github.swent.echo.ExcludeFromJacocoGeneratedReport
+import com.github.swent.echo.R
+import com.github.swent.echo.data.model.Section
+import com.github.swent.echo.data.model.SectionEPFL
+import com.github.swent.echo.data.model.Semester
+import com.github.swent.echo.data.model.SemesterEPFL
+import com.github.swent.echo.data.model.Tag
+import com.github.swent.echo.ui.navigation.NavigationActions
+
+/**
+ * A composable function that displays the UI for creating a user profile.
+ *
+ * @param sectionList The list of sections to be displayed in the dropdown menu.
+ * @param semList The list of semesters to be displayed in the dropdown menu.
+ * @param tagList The list of tags to be displayed as chips.
+ */
+@Composable
+fun ProfileCreationUI(
+    sectionList: List<Section>,
+    semList: List<Semester>,
+    tagList: List<Tag>,
+    // viewModel: createProfileViewModel = hiltViewModel(),
+    navAction: NavigationActions
+) {
+    Box(modifier = Modifier.fillMaxSize().padding(16.dp), contentAlignment = Alignment.Center) {
+        Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
+            var firstName by remember { mutableStateOf("") }
+            var lastName by remember { mutableStateOf("") }
+
+            // Back button
+            Icon(
+                Icons.Default.ArrowBack,
+                contentDescription = "go back",
+                modifier =
+                    Modifier.size(35.dp).clickable(onClick = { navAction.goBack() }).testTag("Back")
+            )
+
+            // First name and last name fields
+            OutlinedTextField(
+                value = firstName,
+                onValueChange = { firstName = it },
+                modifier = Modifier.testTag("FirstName"),
+                label = { Text(text = stringResource(id = R.string.profile_creation_first_name)) }
+            )
+
+            Spacer(modifier = Modifier.height(5.dp))
+
+            OutlinedTextField(
+                value = lastName,
+                onValueChange = { lastName = it },
+                modifier = Modifier.testTag("LastName"),
+                label = { Text(text = stringResource(id = R.string.profile_creation_last_name)) }
+            )
+
+            Spacer(modifier = Modifier.height(5.dp))
+
+            // Section and semester dropdown menus
+            DropDownListFunctionWrapper(sectionList, R.string.profile_creation_section)
+            Spacer(modifier = Modifier.height(5.dp))
+            DropDownListFunctionWrapper(semList, R.string.profile_creation_semester)
+
+            Spacer(modifier = Modifier.height(10.dp))
+
+            // Tags
+            Text(
+                stringResource(id = R.string.profile_creation_tags),
+                fontSize = 20.sp,
+                fontWeight = FontWeight.Bold
+            )
+
+            Spacer(modifier = Modifier.height(10.dp))
+
+            Row() {
+                for (tag in tagList) {
```

Comment:
You can use Yoan implementation of tag selection. You need to rebase your branch and then use TagSelectionDialog. Ask Yoan if you need help on how to use it